exercises/knapsack/canonical-data.json

```
@@ -0,0 +1,115 @@
+{
+  "exercise": "knapsack",
+  "version": "1.0.0",
+  "comments": [
+    "Depending on the language, the input type can be modified"
+  ],
+  "cases": [
+    {
+      "description": "no items",
+      "property": "maximumValue",
+      "input": {
+        "maximumWeight": 100,
+        "items": []
+      },
+      "expected": 0
+    },
+    {
+      "description": "one item, too heavy",
+      "property": "maximumValue",
+      "input": {
+        "maximumWeight": 10,
+        "items": [
+          [100, 1]
```

Excellent work so far.

I've pushed a few changes related to the formatting of the README file, adjusting for some changes that were somehow marked as done and then lost (word-wrapping, present tense, some commas, etc.), and moved a test case a little down.

I have one proposal for a change left: Since the weights and the values were already mixed up once, perhaps this is a good case for naming the pairs, i.e.

```suggestion
          { "weight": 100, "value": 1 }
```

and so on for the remainder of weight/value pairs.

Naturally, tracks may convert this however they like depending on how pairs are best modelled in a given language.

Feel free to discuss or object to this proposal; my review can be dismissed on the account of making a decision here.

---

I agree with this as well. 

A few times while generating some of the test data, I also had to check the description again to remind myself which is the weight/value. This will help make it much more explicit.

I was worried that this may be hard to implement in certain languages. I will change and push it in soon.